registration/registration_test.go

```diff
@@ -415,8 +521,101 @@ func Test_CheckCertificate(t *testing.T) {
 				Data:      data,
 				Signature: ed25519.Sign(wrongPrivate, data),
 			}
-			_, _, err = r.Submit(context.Background(), challenge, nodeID, 0, r.PowParams(), certificate, time.Time{})
+			_, _, err = r.Submit(
+				context.Background(), challenge,
+				pubKeyHint, nodeID, 0, r.PowParams(), certificate, time.Time{})
 			require.ErrorIs(t, err, registration.ErrInvalidCertificate)
 		})
 	})
 }
+
+func generateTestKeyPair() ([]byte, []byte, error) {
+	pubKey, priv, err := ed25519.GenerateKey(nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	return []byte(base64.StdEncoding.EncodeToString(pubKey)), priv, nil
+}
+
+func createTestKeyFile(t *testing.T, dir, name string, data []byte) string {
+	path := filepath.Join(dir, name)
+	if err := os.WriteFile(path, data, 0o644); err != nil {
+		t.Fatalf("failed to write key file: %v", err)
+	}
+	return path
+}
+
+func TestLoadTrustedKeys(t *testing.T) {
+	genesis := time.Now()
+
+	roundCfg := server.RoundConfig{
+		EpochDuration: time.Hour,
+		PhaseShift:    time.Second,
+	}
+
+	workerSvc := mocks.NewMockWorkerService(gomock.NewController(t))
+
+	const keysNum = 3
+
+	dir := t.TempDir()
+	t.Cleanup(func() { os.RemoveAll(dir) })
```

Comment:
NIT: cleanup happens automatically 🙂 

```suggestion
	dir := t.TempDir()
```